multiversx_sdk/multisig/resources.py

```diff
@@ -60,6 +62,54 @@ def __init__(self, token_identifier: str, token_nonce: int, amount: int) -> None
         )
 
 
+class UserRole(Enum):
+    NONE = 0
+    PROPOSER = 1
+    BOARD_MEMBER = 2
+
+
+class ActionFullInfo:
+    def __init__(self, action_id: int, group_id: int, action_data: "Action", signers: list[Address]) -> None:
+        self.action_id = action_id
+        self.group_id = group_id
+        self.action_data = action_data
+        self.signers = signers
+
+    @classmethod
+    def new_from_object(cls, object: Any) -> "ActionFullInfo":
+        signers = [Address(value, LibraryConfig.default_address_hrp) for value in object.signers]
+        action_data = create_action_from_object(object.action_data)
+        return cls(object.action_id, object.group_id, action_data, signers)
+
+    def __repr__(self) -> str:
+        return f"ActionFullInfo(action_id={self.action_id}, group_id={self.group_id}, action_data={self.action_data}, signers={self.signers})"
+
+
+def create_action_from_object(object: Any) -> "Action":
+    discriminant = int(object)
```

Comment:
Somehow hard to understand, for the readers. Maybe add a comment to explain why?

Reply:
added a comment